xds/internal/balancer/clusterimpl/balancer_test.go

```diff
@@ -470,17 +465,7 @@ func (s) TestClusterNameInAddressAttributes(t *testing.T) {
 
 	b.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Ready})
 	// Test pick with one backend.
-	p1 := <-cc.NewPickerCh
-	const rpcCount = 20
-	for i := 0; i < rpcCount; i++ {
-		gotSCSt, err := p1.Pick(balancer.PickInfo{})
-		if err != nil || !cmp.Equal(gotSCSt.SubConn, sc1, cmp.AllowUnexported(testutils.TestSubConn{})) {
-			t.Fatalf("picker.Pick, got %v, %v, want SubConn=%v", gotSCSt, err, sc1)
-		}
-		if gotSCSt.Done != nil {
-			gotSCSt.Done(balancer.DoneInfo{})
-		}
-	}
+	cc.WaitForRoundRobinPicker(ctx, sc1)
```

Comment:
Same question. This would pass if no round robin picker received.

Reply:
As above.